### PR TITLE
CN-7098: Biometrics implementation

### PR DESCRIPTION
**Pull request by CrnogoracIvan**
## Checklist (Mandatory):

- [x] I have named the PR correctly (Check PR1)
- [x] Check the commit’s or even all commits message styles matches our requested structure. (issue number, etc.) (Check CR10)
- [x] I have tested new code locally (Check CR3)
- [x] I have done a smoke test of the larger feature locally (Check CR3)
- [x] I have added at least two reviewers (Check PR5)
- [x] I have assigned the PR to myself (Check PR4)
- [x] I have merged develop (parent) branch into my feature branch and resolved all conflicts (Check PR7)

## Checklist (Optional):

- [ ] I have added necessary documentation, including instructions how to test (Check CR8)
- [ ] I have added tests that prove my fix/implementation is effective or that my feature works
- [ ] I have written the code according to the adopted style guide? (Check CR7)

## Additional notes

> What does this implement/fix? Explain your changes. (optional)
> Are there any scripts that need running (eg. yarn install)?
> Are there any edge cases that need checking?
> Are there any specific test instructions?

[MVPWorkshop Pull Request & Code Review Guidelines](https://docs.google.com/document/d/1TcuU1JBhlGZLtpy0osSNinXQ2piP_yrDuRcXdqaOS_U/edit#heading=h.5b6kpdr4btq8)

### CODE REVIEW

File: app/components/atoms/UserActionsLog/UserActionsLog.js [hunk 1]
```diff
@@ -194,6 +194,20 @@ const getData = item => {
       action: "Withdrawal address change",
     };
   }
+  if (item.action === "biometrics-activated") {
```
Comment (fjovakaric) [inline app/components/atoms/UserActionsLog/UserActionsLog.js:197]: move to constants?

File: app/components/atoms/UserActionsLog/UserActionsLog.js [hunk 2]
```diff
@@ -194,6 +194,20 @@ const getData = item => {
...
+    return {
+      name: "Fingerprint",
+      color: `${getColor(COLOR_KEYS.PRIMARY_BUTTON)}`,
+      action: "Biometrics activated",
+    };
+  }
+  if (item.action === "biometrics-deactivated") {
```
Comment (fjovakaric) [inline app/components/atoms/UserActionsLog/UserActionsLog.js:204]: move to constants?

File: app/components/atoms/UserActionsLog/UserActionsLog.js [hunk 3]
```diff
@@ -194,6 +194,20 @@ const getData = item => {
       action: "Withdrawal address change",
     };
   }
+  if (item.action === "biometrics-activated") {
+    return {
+      name: "Fingerprint",
+      color: `${getColor(COLOR_KEYS.PRIMARY_BUTTON)}`,
```
Comment (fjovakaric) [inline app/components/atoms/UserActionsLog/UserActionsLog.js:200]: don't need `${}`

File: app/components/atoms/UserActionsLog/UserActionsLog.js [hunk 4]
```diff
@@ -194,6 +194,20 @@ const getData = item => {
...
+      action: "Biometrics activated",
+    };
+  }
+  if (item.action === "biometrics-deactivated") {
+    return {
+      name: "Fingerprint",
+      color: `${getColor(COLOR_KEYS.PRIMARY_BUTTON)}`,
```
Comment (fjovakaric) [inline app/components/atoms/UserActionsLog/UserActionsLog.js:207]: don't need `${}`

File: app/components/screens/BiometricActivation/BiometricActivation.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+  static navigationOptions = () => ({
+    headerSameColor: false,
+    transparent: true,
+    hideBack: true,
+    gesturesEnabled: false,
+  });
+  onPressEnableBiometric = async () => {
+    const { actions, biometrics } = this.props;
+    // const biometryTypeCopy =
```
Comment (fjovakaric) [inline app/components/screens/BiometricActivation/BiometricActivation.js:37]: remove comments

File: app/components/screens/BiometricAuthentication/BiometricAuthentication.js [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+      actions.navigateTo(SCREENS.VERIFY_PROFILE, {
+        onSuccess: async () => {
+          try {
+            await createBiometricsKey(publicKey => {
+              actions.activateBiometrics(publicKey, biometrics.biometryType);
+            });
+            actions.resetToScreen(SCREENS.BIOMETRICS_AUTHENTICATION);
+            actions.showMessage("success", enableBiometricsText);
+          } catch (e) {
+            mixpanelAnalytics.logError(
```
Comment (fjovakaric) [inline app/components/screens/BiometricAuthentication/BiometricAuthentication.js:93]: boom!

File: app/redux/biometrics/biometricsActions.js [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+    pin: formData.pin,
+    twoFactorCode: formData.code,
+  };
+  return async dispatch => {
+    try {
+      dispatch(startApiCall(API.DEACTIVATE_BIOMETRICS));
+      await biometricsService.deactivateBiometrics(verification);
+
+      dispatch({
+        type: ACTIONS.UPDATE_USER_BIOMETRICS_STATUS_SUCCESS,
```
Comment (fjovakaric) [inline app/redux/biometrics/biometricsActions.js:112]: DEACTIVATE_BIOMETRICS_SUCCESS

File: app/utils/biometrics-util.js [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+    if (onSuccess) onSuccess(res.publicKey);
+  } catch (e) {
+    mixpanelAnalytics.logError("createBiometricsKey", e);
+  }
+}
+
+/**
+ * Check if biometrics key is already created
+ */
+async function checkBiometricsKey() {
```
Comment (fjovakaric) [inline app/utils/biometrics-util.js:44]: try/catch

**Review by fjovakaric — CHANGES_REQUESTED** [review]

File: app/components/screens/BiometricActivation/BiometricActivation.js (hunk 5 above)
Comment (CrnogoracIvan) [reply app/components/screens/BiometricActivation/BiometricActivation.js:37]: This screen will be deleted.

File: app/components/atoms/UserActionsLog/UserActionsLog.js (hunk 4 above)
Comment (CrnogoracIvan) [reply app/components/atoms/UserActionsLog/UserActionsLog.js:207]: Done, fixed on other too.

File: app/components/screens/BiometricAuthentication/BiometricAuthentication.js (hunk 6 above)
Comment (CrnogoracIvan) [reply app/components/screens/BiometricAuthentication/BiometricAuthentication.js:93]: 🚀

File: app/redux/biometrics/biometricsActions.js (hunk 7 above)
Comment (CrnogoracIvan) [reply app/redux/biometrics/biometricsActions.js:113]: Done

File: app/utils/biometrics-util.js (hunk 8 above)
Comment (CrnogoracIvan) [reply app/utils/biometrics-util.js:44]: Done

**Review by fjovakaric — APPROVED** [review]

**Pull request merged**